pkg/providers/vsphere/session/session_vm_update.go

```diff
@@ -1099,3 +1108,38 @@ func (s *Session) UpdateVirtualMachine(
 
 	return err
 }
+
+func updateVMPaused(
+	vmCtx context.VirtualMachineContext,
+	moVM *mo.VirtualMachine) bool {
+
+	vm := vmCtx.VM
+	var adminPaused, devopsPaused bool
+
+	for i := range moVM.Config.ExtraConfig {
+		if o := moVM.Config.ExtraConfig[i].GetOptionValue(); o != nil {
+			if o.Key == vmopv1.PauseVMExtraConfigKey && o.Value == constants.ExtraConfigTrue {
```

Comment:
We can break whenever/if we find the key and `adminPaused` can be set depending on the value.